### ISSUE TITLE
Refactor tests to be more independent

### ISSUE THREAD

**Issue by rtholmes** [ubccpsc/classy#65]
Tests shouldn't depend on state from other tests. E.g., the DB should be wiped in each `before` block.

**Comment by jopika** [comment]
Would this increase test times significantly?

Also, I found a really nice feature that allows sub-grouping of tests inside a test file, which might be useful if you have tests that are highly dependant on eachother (perhaps when one test generates a result the other might use).

**Comment by jopika** [comment]
I agree that we should probably avoid state depenancy when it comes to the the tests, especially between test files.

**Comment by rtholmes** [comment]
While it would involve more setup and teardown of objects, I think it would enable parallelization (since each test suite could be run independently on different hosts) so it could potentially get faster.
